markwon/src/main/java/ru/noties/markwon/renderer/SpannableMarkdownVisitor.java

```diff
@@ -465,23 +481,13 @@ public void visit(Link link) {
         setSpan(length, factory.link(theme, destination, configuration.linkResolver()));
     }
 
-    private void setSpan(int start, @Nullable Object span) {
-        if (span != null) {
-
-            final int length = builder.length();
-
-            SpannableBuilder.setSpans(
-                    builder,
-                    span,
-                    start,
-                    length
-            );
-        }
+    private void setSpan(int start, @Nullable Object spans) {
+        builder.setSpan(spans, start, builder.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
     }
 
     private void newLine() {
         if (builder.length() > 0
-                && '\n' != builder.lastChar()) {
+                && '\n' != builder.charAt(builder.length() - 1)) {
```

Comment:
We _could_ also have one call to length by storing it a variable

Reply:
true that! 👍